Extensions/Lighting/lightruntimeobject-pixi-renderer.js

```diff
@@ -94,4 +115,77 @@ gdjs.LightRuntimeObjectPixiRenderer.prototype.updateVertexBuffer = function () {
         this._object.y,
       ])
     );
+//  }
+};
+
+gdjs.LightRuntimeObjectPixiRenderer.prototype.raycastTest = function () {
```

Comment:
@4ian this doesn't really work. I mostly get empty arrays and I don't understand why is that so. The expected return value should be at least the vertices of the mesh (`this._object.getHitBoxes().vertices`). Can you take a look at this function? There's definitely something wrong here :/

Reply:
I guess I'm doing something wrong with _`closeX`, `closeY, farX and farY`_ 🤔 Can you explain the return value (`result`) of _`gdjs.Polygon.raycastTest()`_ ?

Reply:
When you launch a raycast against a polygon, the result is:
- was there even a collision between the ray and the polygon? (`collision` boolean)
- coordinate of the closest point between the ray and the polygon (closeX, closeY)
- the squared distance of this point (`closeSqDist`)
- coordinate of the farthest point between the ray and the polygon (closeX, closeY). You can picture it as the point behind the polygon, if you think of the ray as continue through the polygon:
- the squared distance of this point (`farSqDist`)

![image](https://user-images.githubusercontent.com/1280130/84707479-acd65f80-af56-11ea-9fcf-9523981ddb80.png)

In our algorithm, we don't need the farX, farY point :)